color.cpp

```diff
@@ -29,62 +29,109 @@ int get_opaqueness(const color &c) noexcept { return c.get_opaqueness(); }
 {
-    std::string msg = "[";
-    msg+=std::to_string(in_color.get_red())+", ";
-    msg+=std::to_string(in_color.get_green())+", ";
-    msg+=std::to_string(in_color.get_blue())+", ";
-    msg+=std::to_string(in_color.get_opaqueness())+"]";
-    return msg;
+  return color(255, 0, 0, 255);
 }
 
+color create_green_color()
+{
+  return color(0, 255, 0, 255);
+}
 
-void test_color()
+color create_blue_color()
 {
-    #ifndef NDEBUG // no tests is release
-    // Color is white by default
-    {
-        const color c;
-        assert(c.get_red() == 255);
-        assert(c.get_green() == 255);
-        assert(c.get_blue() == 255);
-        assert(c.get_opaqueness() == 255);
-    }
-    // Test get functions
-    {
-        const color c1{0,21,255,10};
-        assert(get_redness(c1)==0);
-        assert(get_greenness(c1)==21);
-        assert(get_blueness(c1)==255);
-        assert(get_opaqueness(c1)==10);
-    }
-    // operator==
+  return color(0, 0, 255, 255);
+}
+
+double calc_hue(const color &c)
+{
+  double r = c.get_red()   / 255.0;
+  double g = c.get_green() / 255.0;
+  double b = c.get_blue()  / 255.0;
+  double max_val = std::max(std::max(r, g), b);
+  double min_val = std::min(std::min(r, g), b);
+  double hue;
+
+  if (max_val == min_val)
     {
-        const color c;
-        const color c1;
-        assert(c == c1);
+      hue = 0.0;
     }
-
-//     operator!=
+  else if (max_val == r)
     {
-      const color red{255, 0, 0};
-      const color green{0, 255, 0};
-      assert(red != green);
+      hue = 0.0 + (g - b) / (max_val - min_val);
+      if (hue < 0.0)
+        {
+          hue += 6.0;
+        }
     }
-    // operator<<
+  else if (max_val == g)
     {
-        std::stringstream s;
-        const color c;
-        s << c;
+      hue = 2.0 + (b - r) / (max_val - min_val);
     }
+  else
     {
-        const color c;
-        std::cout << c;
+      hue = 4.0 + (r - g) / (max_val - min_val);
     }
-  //#define FIX_ISSUE_322
-  #ifdef FIX_ISSUE_322
+  return hue * 60.0;
+}
+
+//convert color to string in debugging purposes
+std::string to_str(const color& in_color) noexcept
+{
+  std::string msg = "[";
+  msg+=std::to_string(in_color.get_red())+", ";
+  msg+=std::to_string(in_color.get_green())+", ";
+  msg+=std::to_string(in_color.get_blue())+", ";
+  msg+=std::to_string(in_color.get_opaqueness())+"]";
+  return msg;
+}
+
+
+void test_color()
+{
+#ifndef NDEBUG // no tests is release
+  // Color is white by default
+  {
+    const color c;
+    assert(c.get_red() == 255);
+    assert(c.get_green() == 255);
+    assert(c.get_blue() == 255);
+    assert(c.get_opaqueness() == 255);
+  }
+  // Test get functions
+  {
+    const color c1{0,21,255,10};
+    assert(get_redness(c1)==0);
+    assert(get_greenness(c1)==21);
+    assert(get_blueness(c1)==255);
+    assert(get_opaqueness(c1)==10);
+  }
+  // operator==
+  {
+    const color c;
+    const color c1;
+    assert(c == c1);
+  }
+
+  //     operator!=

```

Comment:
No idea why so many spaces between `//` and `operator==`, no worries however :-)

Reply:
Not sure either, I didn't write these functions, just fixed the indentation I believe.
I will correct this in my next merge 👍